lib/SILOptimizer/Transforms/SILLowerAggregateInstrs.cpp

```diff
@@ -109,17 +112,23 @@ static bool expandCopyAddr(CopyAddrInst *CA) {
     //   retain_value %new : $*T
     IsTake_t IsTake = CA->isTakeOfSrc();
     if (IsTake_t::IsNotTake == IsTake) {
-      TL.emitLoweredCopyValue(Builder, CA->getLoc(), New,
-                              TypeLowering::LoweringStyle::Deep);
+      if (expand)
+        TL.emitLoweredCopyValue(Builder, CA->getLoc(), New,
+                                TypeLowering::LoweringStyle::Deep);
+      else
+        TL.emitCopyValue(Builder, CA->getLoc(), New);
```

Comment:
Actually, LoweringStyle is a 2 case enum! So you could do LoweringStyle(shouldExpand). My only concern about that is if we add an additional LoweringStyle.